src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -80,33 +80,32 @@ object Metric {
   }
 
 
-  /** Construct a metric of the given type.  The arguments given should be a sequence of tuples: the name of the field
-    * and value as a string.
-    * @deprecated This method is very slow since it inspects the target type via reflection for each instance
-    *             created.  Use [[read()]] instead.
-    *
+  /** Construct one or metrics of the given type.  There should be a sequence of tuples per metric to be built.  The
+    * The arguments for a single metric given should be a sequence of tuples: the name of the field and value as a string.
     * */
-  @deprecated("Very slow.  Use read() instead.", since="0.3.1")
-  def build[T <: Metric](args: Seq[(String, String)])(implicit tt: ru.TypeTag[T]): T = {
+  def build[T <: Metric](metricArgs: TraversableOnce[(String, String)]*)(implicit tt: ru.TypeTag[T]): Seq[T] = {
     val clazz             = ReflectionUtil.typeTagToClass[T]
     val reflectiveBuilder = new ReflectiveBuilder(clazz)
-    // set the arguments
-    args.foreach { case (name, value) =>
-      reflectiveBuilder.argumentLookup.forField(name) match {
-        case Some(arg) =>
-          val argumentValue = ReflectionUtil.constructFromString(arg.argumentType, arg.unitType, value) match {
-            case Success(v) => v
-            case Failure(thr) => throw thr
-          }
-          arg.value_=(argumentValue)
-        case None =>
-          throw new IllegalArgumentException(s"The class '${clazz.getSimpleName}' did not have a field with name '$name'.")
+
+    metricArgs.map { args =>
+      // set the arguments
+      args.foreach { case (name, value) =>
+        reflectiveBuilder.argumentLookup.forField(name) match {
+          case Some(arg) =>
+            val argumentValue = ReflectionUtil.constructFromString(arg.argumentType, arg.unitType, value) match {
+              case Success(v) => v
+              case Failure(thr) => throw thr
+            }
+            arg.value_=(argumentValue)
+          case None =>
+            throw new IllegalArgumentException(s"The class '${clazz.getSimpleName}' did not have a field with name '$name'.")
+        }
```

Comment:
Just a thought, but what if not all the arguments are given in a sequence of tuples, will we use the value from the previous sequence?

```
scala> class Foo(str: String, int: Int = 4)
scala> Metric.build[Foo](Seq(("str", "value1"), ("int", 2)), Seq(("str", "value2")))
res0: Seq(Foo("str", 2), Foo("value2", 2)))
```
but the result should be: `Foo("str", 2), Foo("value2", 4)))

Reply:
I agree, this seems like a problem.  For loading from a file, we'll have the same set of columns each time. But for any other use case that can't be guaranteed.

Reply:
Fixed